api/tests/src/api/users/test_user_save_opportunity_post.py

```diff
@@ -0,0 +1,97 @@
+import uuid
+
+from src.auth.api_jwt_auth import create_jwt_for_user
+from src.db.models.user_models import UserSavedOpportunity
+from tests.src.db.models.factories import OpportunityFactory, UserFactory
+
+
+def test_user_save_opportunity_post_unauthorized_user(app, enable_factory_create, db_session):
+    # Create a user and get their token
+    user = UserFactory.create()
+    token, _ = create_jwt_for_user(user, db_session)
+    db_session.commit()
```

Comment:
We could use a test utility that is like "get_test_user" that we'll eventually be able to pass things like roles into for a given unit test. Might be worth looking into setting that up.

Reply:
I did abstract this slightly to remove duplication. I suppose the fixture I added could be added to conftest?